storage/cache/subtree_cache.go

```diff
@@ -39,52 +52,75 @@ type Suffix struct {
 	// bits is the number of bits in the node ID suffix.
 	bits byte
 	// path is the suffix itself.
-	path byte
+	path []byte
 }
 
 func (s Suffix) serialize() string {
-	r := make([]byte, 2)
+	r := make([]byte, 1, 1+(len(s.path)))
 	r[0] = s.bits
-	r[1] = s.path
+	r = append(r, s.path...)
 	return base64.StdEncoding.EncodeToString(r)
 }
 
-const (
-	// strataDepth is the depth of Subtree.
-	strataDepth = 8
-)
-
 // NewSubtreeCache returns a newly intialised cache ready for use.
 // populateSubtree is a function which knows how to populate a subtree's
 // internal nodes given its leaves, and will be called for each subtree loaded
 // from storage.
 // TODO(al): consider supporting different sized subtrees - for now everything's subtrees of 8 levels.
-func NewSubtreeCache(populateSubtree storage.PopulateSubtreeFunc) SubtreeCache {
+func NewSubtreeCache(strata []int, populateSubtree storage.PopulateSubtreeFunc) SubtreeCache {
+	// TODO(al): pass this in
+	maxTreeDepth := 256
+	// Precalculate strata information based on the passed in strata depths:
+	sInfo := make([]strataInfo, 0, maxTreeDepth/8)
+	t := 0
+	for _, s := range strata {
+		if s%8 != 0 {
```

Comment:
this looks like it can go wrong if the caller passes in duplicate / inconsistent strata entries